oak-store-spi/src/main/java/org/apache/jackrabbit/oak/spi/state/NodeStateUtils.java

```diff
@@ -53,12 +52,7 @@ public static boolean isHidden(@NotNull String name) {
      * @return true if one of the nodes is hidden
      */
     public static boolean isHiddenPath(@NotNull String path) {
-        for (String n : PathUtils.elements(path)) {
-            if (isHidden(n)) {
-                return true;
-            }
-        }
-        return false;
+        return (!path.isEmpty() && path.charAt(0) == ':') || path.contains("/:");
```

Comment:
There's one behaviour change introduced by this PR - isHiddenPath no longer checks if he path is valid or not. Earlier this used to happen when calling PathUtils.elements.

This is not documented for this method, so in theory (might) be ok as well. Otherwise the change looks ok to me.

Reply:
Good point. But the check in `PathUtils.elements(path)` is done by an `assert`. And very likely asserts will be disabled in production (the default in the JVM is disabled). And in any case, it is bad practice to rely on asserts in production. So I think it is ok to change the behavior.